File: Database/Assembly_SRPG_JP/TowerParam.cs
// Decompiled with JetBrains decompiler
// Type: SRPG.TowerParam
// Assembly: Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
// MVID: 85BFDF7F-5712-4D45-9CD6-3465C703DFDF
// Assembly location: S:\Desktop\Assembly-CSharp.dll

namespace SRPG
{
  public class TowerParam
  {
    public string iname;
    public string name;
    public string expr;
    public string banr;
    public string bg;
    public string floor_bg_open;
    public string floor_bg_close;
    public short unit_recover_minute;
    public short unit_recover_coin;
    public string prefabPath;
    public string eventURL;
    public bool can_unit_recover;
    public bool is_down;
    public bool is_view_ranking;
    public short unlock_level;
    public string unlock_quest;
    public bool is_unlock;
    public string URL;
    public short floor_reset_coin;
    public string score_iname;

    public void Deserialize(JSON_TowerParam json)
    {
      if (json == null)
        throw new InvalidJSONException();
      this.iname = json.iname;
      this.name = json.name;
      this.expr = json.expr;
      this.banr = json.banr;
      this.prefabPath = json.item;
      this.bg = json.bg;
      this.floor_bg_open = json.floor_bg_open;
      this.floor_bg_close = json.floor_bg_close;
      this.can_unit_recover = json.can_unit_recover == (byte) 1;
      this.unit_recover_minute = json.unit_recover_minute;
      this.unit_recover_coin = json.unit_recover_coin;
      this.eventURL = json.eventURL;
      this.is_down = json.is_down > (byte) 0;
      this.is_view_ranking = json.is_view_ranking > (byte) 0;
      this.unlock_level = json.unlock_level;
      this.unlock_quest = json.unlock_quest;
      this.URL = json.url;
      this.floor_reset_coin = json.floor_reset_coin;
      this.score_iname = json.score_iname;
    }
  }
}

File: Database/genConvert/TowerParam.py
def TowerParam(json):
    this={}#TowerParamjson)
    #if(json==null)
    #thrownewInvalidJSONException()
    if 'iname' in json:
        this['iname'] = json['iname']
    if 'name' in json:
        this['name'] = json['name']
    if 'expr' in json:
        this['expr'] = json['expr']
    if 'banr' in json:
        this['banr'] = json['banr']
    if 'item' in json:
        this['prefabPath'] = json['item']
    if 'bg' in json:
        this['bg'] = json['bg']
    if 'floor_bg_open' in json:
        this['floor_bg_open'] = json['floor_bg_open']
    if 'floor_bg_close' in json:
        this['floor_bg_close'] = json['floor_bg_close']
    if 'can_unit_recover' in json:
        this['can_unit_recover'] = json['can_unit_recover']==1
    if 'unit_recover_minute' in json:
        this['unit_recover_minute'] = json['unit_recover_minute']
    if 'unit_recover_coin' in json:
        this['unit_recover_coin'] = json['unit_recover_coin']
    if 'eventURL' in json:
        this['eventURL'] = json['eventURL']
    if 'is_down' in json:
        this['is_down'] = json['is_down']>0
    if 'is_view_ranking' in json:
        this['is_view_ranking'] = json['is_view_ranking']>0
    if 'unlock_level' in json:
        this['unlock_level'] = json['unlock_level']
    if 'unlock_quest' in json:
        this['unlock_quest'] = json['unlock_quest']
    if 'url' in json:
        this['URL'] = json['url']
    if 'floor_reset_coin' in json:
        this['floor_reset_coin'] = json['floor_reset_coin']
    if 'score_iname' in json:
        this['score_iname'] = json['score_iname']
return this
